atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -185,8 +189,17 @@ private void addSnapshotListener() {
                   .build()));
     }
 
-    final var snapshotChunk = new SnapshotChunkImpl();
-    snapshotChunk.wrap(new UnsafeBuffer(request.data()), 0, request.data().capacity());
+    SnapshotChunk snapshotChunk;
+    try {
+      final var snapshotChunkImpl = new SnapshotChunkImpl();
+      snapshotChunkImpl.wrap(new UnsafeBuffer(request.data()), 0, request.data().capacity());
```

Comment:
Suggestion: you can use `snapshotChunkImpl.tryWrap`

Reply:
The way tryWrap is implemented would most likely work here, though it would also normally reject messages with the wrong schema ID or versions...still works but I feel for the wrong reason (as it doesn't explicitly fail if it cannot be parsed, just if some header properties are wrong). wdyt?

Reply:
Ya. In this case, it does not make much difference. Might be useful for future-proofing.